File: frontend/src/types/qianchuanCollection.ts
export interface CollectionPersona {
  name: string;
  script_count: number;
}

export interface CollectionScript {
  id: number;
  pool: 'global' | 'persona';
  persona_name: string | null;
  title: string;
  content: string;
  likes: number | null;
  source: string | null;
  source_account: string | null;
  script_date: string | null;
  created_at: string;
}

export interface ScriptListResponse {
  scripts: CollectionScript[];
  total: number;
  page: number;
  page_size: number;
}

export interface CreateScriptBody {
  pool: 'global' | 'persona';
  persona_name?: string;
  title: string;
  content: string;
  likes?: number;
  source?: string;
  source_account?: string;
  script_date?: string;
}

File: backend/app/models/qianchuan_collection.py
from sqlalchemy import Column, Integer, String, Text, Boolean, Date, TIMESTAMP, func
from app.core.database import Base


class QianchuanCollectionPersona(Base):
    """千川爆文合集 — 达人分组表"""
    __tablename__ = "qianchuan_collection_personas"

    id         = Column(Integer, primary_key=True, autoincrement=True)
    name       = Column(String(100), nullable=False, unique=True)
    is_deleted = Column(Boolean, nullable=False, default=False)
    created_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    updated_at = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())


class QianchuanCollectionScript(Base):
    """千川爆文合集 — 脚本表"""
    __tablename__ = "qianchuan_collection_scripts"

    id             = Column(Integer, primary_key=True, autoincrement=True)
    pool           = Column(String(20), nullable=False, default="global")
    persona_name   = Column(String(100), nullable=True)
    title          = Column(String(200), nullable=False)
    content        = Column(Text, nullable=False)
    likes          = Column(Integer, nullable=True)
    source         = Column(String(100), nullable=True)
    source_account = Column(String(100), nullable=True)
    script_date    = Column(Date, nullable=True)
    is_deleted     = Column(Boolean, nullable=False, default=False)
    created_at     = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    updated_at     = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
